src/pytorch_lightning/core/module.py

```diff
@@ -1786,6 +1792,66 @@ def _verify_is_manual_optimization(self, fn_name: str) -> None:
                 " set model property `automatic_optimization` as False"
             )
 
+    def compile(
+        self,
+        backend: Union[str, Callable],
+        fullgraph: bool = False,
+        dynamic: bool = False,
+        mode: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Optimizes the model using the Dynamo compiler.
+
+        Args:
+            backend (str or Callable): compiler backend to be used. Default: ``"inductor"``.
+                Call ``torch._dynamo.list_backends()`` to get a full list.
+            dynamic (bool): Use dynamic shape tracing. Default: ``False``.
+            fullgraph (bool): Whether it is ok to break model into several subgraphs. Default: ``False``.
+            mode (str): Can be either ``"default"``, ``"reduce-overhead"`` or ``"max-autotune"``.
+                Only relevant for the "inductor" backend.
```

Comment:
```suggestion
            backend: compiler backend to be used.
                Call ``torch._dynamo.list_backends()`` to get a full list.
            dynamic: Use dynamic shape tracing.
            fullgraph: Whether it is ok to break model into several subgraphs.
            mode: Can be either ``"default"``, ``"reduce-overhead"`` or ``"max-autotune"``.
                Only relevant for the "inductor" backend.
```
already in the method typing, so in docs too